cookbooks/docker/recipes/default.rb

```diff
@@ -1,33 +1,27 @@
-# Recipe for Docker
+# Recipe for Docker Engine and Docker Compose
 
-# Install prerequisite packages
-apt_update "update-apt-cache" do
-  action :update
-end
-
-apt_package "prerequisites-linux" do
-  package_name ["curl", "linux-image-extra-" << `uname -r`, "linux-image-extra-virtual"]
-end
-
-apt_package "prerequisites-apt" do
-  package_name ["apt-transport-https", "ca-certificates"]
+# Add APT key and repository
+apt_repository "docker" do
+  uri "https://apt.dockerproject.org/repo"
+  distribution "ubuntu-#{node["docker"]["ubuntu_release"]}"
+  components ["main"]
+  key "https://yum.dockerproject.org/gpg"
+  action :add
 end
 
 
-# Add APT key
-execute "add-apt-key" do
-  command "curl -fsSL https://yum.dockerproject.org/gpg | sudo apt-key add -"
+# Install Docker Engine and prerequisite packages
+apt_package "docker-engine" do
+  package_name [
```

Comment:
You are right, that will provide me with more flexibility in case Docker's dependencies change in the future. Will change this, thanks. :smiley_cat: